File: PigSim/PigTester.java



public class PigTester {

    public static void main(String[] args) {
        int computerHoldAmt = Integer.parseInt(args[0]);
        PigPlayer p = new PigPlayer(computerHoldAmt);
        
        int userScore = 0;
        int roundCount = 1;

        while (p.getScore() < 35 && userScore < 35) {
            // You'll call, e.g., p.getScore() to get the computers score

            System.out.printf("\nRound %d\n", roundCount);
            System.out.println("------------------------");

            //int turnScore = PigFunctions.takeTurn(computerHoldAmt);
            p.takeTurn();
            // You'll call p.takeTurn()
           
            System.out.printf("The computer scored %d. ", p.getScore());
            System.out.println("Now it's your turn...\n");

            int turnScore = PigFunctions.takeTurnInteractive();
            userScore += turnScore;

            System.out.printf("\nYour score is %d\n", userScore);

            roundCount += 1;
        }
        if (p.getScore() > userScore) {
            System.out.printf("\nYou lose, %d to %d. Sorry!\n", p.getScore(), userScore);
        } else if (userScore > p.getScore()) {
            System.out.printf("\nYou win, %d to %d!!\n", userScore, p.getScore());
        } else {
            System.out.printf("\nIt's a tie, %d to %d.\n", userScore, p.getScore());
        }
    }
}

File: PigSim/pig_tester.py
from sys import argv
import pig_functions

if __name__ == "__main__":
    computerHoldAmt = int(argv[1])
    computerScore = 0
    userScore = 0

    roundCount = 1

    while computerScore < 100 and userScore < 100:
        print("\nRound {} - Computer: {}, User: {}".format(
            roundCount, computerScore, userScore))
        print("-----------------------------------")

        turnScore = pig_functions.takeTurn(computerHoldAmt)
        computerScore += turnScore
        print("The computer scored {}".format(turnScore))
        print("Now it’s your turn ...\n")

        turnScore = pig_functions.takeTurnInteractive()
        userScore += turnScore

        roundCount += 1
        
    if computerScore > userScore:
        print("\nYou lose, {} to {}. Sorry!\n".format(
    computerScore, userScore))
    elif userScore > computerScore:
        print("\nYou win, {} to {}!!\n".format(
        userScore, computerScore))
    else:
        print("\nIt’s a tie, {} to {}.\n".format(
        userScore, computerScore))
    